internal/scaffold/resource.go

```diff
@@ -101,6 +102,56 @@ func (r *Resource) Validate() error {
 	return nil
 }
 
+// ResourceData holds data used to construct a Resource.
```

Comment:
I understand that it will be required/useful just if we decide to call the plugins in the legacy command otherwise the plugin will have the domain, group, version and kind flags and it will use them directly to do the scaffolds without the need to deal with `APIVersion`. So, IMO we need to reach this consensus first.

Reply:
I'm fine with removing `APIVersion`. It is only there for backwards compatibility.

Reply:
I understand that this PR is no longer required. 
we will change the legacy and new one will use the same kb structure instead.